sources/HUBComponentModelBuilderImplementation.m

```diff
@@ -259,19 +269,38 @@ - (BOOL)builderExistsForChildWithIdentifier:(NSString *)identifier
 
 - (id<HUBComponentModelBuilder>)builderForChildWithIdentifier:(NSString *)identifier
 {
-    return [self getOrCreateBuilderForChildWithIdentifier:identifier];
+    return [self getOrCreateBuilderForChildWithIdentifier:identifier groupIdentifier:nil];
+}
+
+- (nullable NSArray<id<HUBComponentModelBuilder>> *)buildersForChildrenInGroupWithIdentifier:(NSString *)groupIdentifier
+{
+    return self.childBuildersByGroupIdentifier[groupIdentifier];
 }
 
 - (void)removeBuilderForChildWithIdentifier:(NSString *)identifier
 {
+    id<HUBComponentModelBuilder> builder = self.childBuilders[identifier];
     self.childBuilders[identifier] = nil;
     [self.childIdentifierOrder removeObject:identifier];
+
+    for (NSString *groupIdentifier in self.childBuildersByGroupIdentifier) {
```

Comment:
Why do you need to iterate through all groups? You already know the group you're looking for (`builder.groupIdentifier`), right?